pkg/admin/facade/user.go

```diff
@@ -109,6 +116,60 @@ func (f *UserFacade) ResetPassword(id string, password string, sendPassword bool
 	return nil
 }
 
+func (f *UserFacade) MarkPasswordAsExpired(id string, isExpired bool) error {
+	err := f.Users.CheckUserAnonymized(id)
+	if err != nil {
+		return err
+	}
+
+	passwordType := apimodel.AuthenticatorTypePassword
+	primaryKind := authenticator.KindPrimary
+	ars, err := f.Authenticators.ListRefsByUsers(
+		[]string{id},
+		&passwordType,
+		&primaryKind,
+	)
+	if err != nil {
+		return err
+	}
+
+	if len(ars) == 0 {
+		return api.ErrNoAuthenticator
```

Comment:
According to the spec, we should introduce a new API error. See docs/specs/admin-api-change-password.md:85